build.gradle

```diff
@@ -22,11 +29,46 @@ dependencies {
     implementation "com.exactpro.th2:codec:5.4.1-dev"
 
     implementation "com.fasterxml.jackson.core:jackson-databind"
-    implementation 'com.github.jsqlparser:jsqlparser:4.7'
+
+    // https://mvnrepository.com/artifact/org.antlr/antlr4
+    antlr 'org.antlr:antlr4:4.13.1'
+    implementation 'org.antlr:antlr4-runtime:4.13.1'
+
+    jmh 'org.openjdk.jmh:jmh-core:0.9'
+    jmh 'org.openjdk.jmh:jmh-generator-annprocess:0.9'
+    jmh 'org.openjdk.jmh:jmh-generator-bytecode:0.9'
+    jmh 'com.github.jsqlparser:jsqlparser:4.7'
 
     compileOnly "com.google.auto.service:auto-service-annotations:1.1.1"
     annotationProcessor "com.google.auto.service:auto-service:1.1.1"
     kapt "com.google.auto.service:auto-service:1.1.1"
 
     testImplementation "com.github.doyaaaaaken:kotlin-csv-jvm:1.9.2"
+}
+
+configurations {
+    compile {
+        // Undo extendsFrom relationship between the 'antlr' configuration and the 'api' configuration
+        // See https://github.com/gradle/gradle/blob/master/subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrPlugin.java#L60
+        extendsFrom = extendsFrom.findAll { it != configurations.antlr }
+    }
+}
+
+jmh {
+    warmupIterations = 1
+    iterations = 3
+    fork = 2
+}
+
+tasks.withType(KotlinCompile).configureEach {
+    dependsOn(tasks.withType(AntlrTask))
+}
+kapt
```

Comment:
Probably, this should be removed

Reply:
Yes, I faced to build problem when extract embedded classes to separate files and I tried to investigate this problem